lib/super_digit.rb

```diff
@@ -1,15 +1,21 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n^2)
 def super_digit(n)
-   
+   return n if n < 10
+
+   return super_digit(n.digits.sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n)
+# Space Complexity - O(n^2)
 def refined_super_digit(n, k)
-    
+  return super_digit(n) if k == 1
+
+  k_n = (n.to_s * k).to_i
+  
+  return super_digit(k_n.digits.sum)
```

Comment:
Consider a dynamic programming solution

```suggestion
  sum = super_digit(n)
  sum *= k
  return super_digit(sum)
```